src/Symfony/Component/Form/Extension/Core/Type/ChoiceType.php

```diff
@@ -160,6 +160,22 @@ public function buildForm(FormBuilderInterface $builder, array $options)
             // transformation is merged back into the original collection
             $builder->addEventSubscriber(new MergeCollectionListener(true, true));
         }
+
+        // To avoid some problem against treating of array (e.g. Array to string conversion),
+        // we have to first ensure the all elements of submitted data ain't an array.
```

Comment:
"To avoid issues when the submitted choices are arrays (i.e. array to string conversions), we have to ensure that all elements of the submitted choice data are strings or null."